paraatm/simulation_method/vcas.py

```diff
@@ -0,0 +1,329 @@
+
+                    ac = self.aircraftInterface.select_aircraft(aclist[0])
+                    ac.setFlight_phase(stat)
+            else:  # the last command
+                stat = ac.getFlight_phase()
+                ac.setCruise_alt_ft(cmd_level)
+                ac.setFlight_phase(11)  # cruise
+                self.simulationInterface.resume(275)
+                # check pause
+                while True:
+                    server_runtime_sim_status = self.simulationInterface.get_runtime_sim_status()
+                    # print(server_runtime_sim_status)
+                    if (server_runtime_sim_status == self.NATS_SIMULATION_STATUS_PAUSE):
+                        break
+
+                ac = self.aircraftInterface.select_aircraft(aclist[0])
+                # curlon = ac.getLongitude_deg()
+                # curlat = ac.getLatitude_deg()
+                ac.setFlight_phase(stat)
+
+        self.simulationInterface.resume()
+        # check end
+        while True:
+            server_runtime_sim_status = self.simulationInterface.get_runtime_sim_status()
+            if (server_runtime_sim_status == self.NATS_SIMULATION_STATUS_ENDED):
+                break
+
+    def write_output(self, filename):
+        """
+        write NATS simulation output to specified file
+        Parameters
+        ----------
+        filename : str
+             directory to output file. Default in simulation function is out put in NATS_HOME with name
+             vcasmodule_xxxxxxxxxxxx.csv (xxxxxxxxxxxx is timestamp * 1000)
+        """
+        self.simulationInterface.write_trajectories(filename)
+
+    def cleanup(self):
+        """
+        clear loaded aircrafts and wind data
+        """
+        self.aircraftInterface.release_aircraft()
+        self.environmentInterface.release_rap()
+
+    def make_anti_model(self):
+        """
+        create a simulation for ignoring each command based on self.command_from
+
+        Returns
+        -------
+        list
+            include: list[0] trajectory data for reference (right simulation)
+                     list[1::] trajectory data for ignoring each command
+            elements in list is an nparray with 2 dimensions recording timestamp, longitude latitude and altitude
+        """
+        cmd_maintain = self.command_from_file()
+        track = self()
+        traj = np.asarray([track.time.values.astype(np.float)//1e9, track['latitude'].values,
+                           track['longitude'].values, track['altitude'].values])
+        models = [traj.T]
+        for i in range(len(cmd_maintain)):
+            new_cmd = cmd_maintain[0:i]
+            track = self(input_cmd=new_cmd)
+            traj = np.asarray([track.time.values.astype(np.float) // 1e9, track['latitude'].values,
+                               track['longitude'].values, track['altitude'].values])
+            models.append(traj.T)
+
+        return models
+
+    def model_update(self, prior=input_prior):
```

Comment:
This creates an error because `input_prior` is not defined.  You probably meant:
```python
   def model_update(self, prior=None):
```